examples/customresources/apis/summationapi/summationapi.go

```diff
@@ -0,0 +1,181 @@
+// Package summation defines a simple number summing service API for demonstration purposes.
+package summationapi
+
+import (
+	"context"
+	"sync"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	pb "go.viam.com/rdk/examples/customresources/apis/proto/api/service/summation/v1"
+	"go.viam.com/utils/rpc"
+
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/subtype"
+	"go.viam.com/rdk/utils"
+	goutils "go.viam.com/utils"
+)
+
+var Subtype = resource.NewSubtype(
+	resource.Namespace("acme"),
+	resource.ResourceTypeService,
+	resource.SubtypeName("summation"),
+)
+
+// Named is a helper for getting the named Summation's typed resource name.
+func Named(name string) resource.Name {
+	return resource.NameFromSubtype(Subtype, name)
+}
+
+func init() {
+	registry.RegisterResourceSubtype(Subtype, registry.ResourceSubtype{
+		Reconfigurable: wrapWithReconfigurable,
+		RegisterRPCService: func(ctx context.Context, rpcServer rpc.Server, subtypeSvc subtype.Service) error {
+			return rpcServer.RegisterServiceServer(
+				ctx,
+				&pb.SummationService_ServiceDesc,
```

Comment:
I think i mentioned it in previous year but we can address this in a follow up PR: internally, this amount of fields to popular feels okay, but externally, for module writers, this feels like a lot versus the necessary bits which are `NewServer`, `SummationService_ServiceDesc`, `RegisterSummationServiceHandlerFromEndpoint`, and RPCClient